### ISSUE TITLE
[iOS][a11y] VoiceOver on UIKitView seems to be broken 

### ISSUE THREAD

**Issue by romgrm** [flutter/flutter#135504]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

Hello !
I've already opened a question on this topic #131780 and by searching more on my side, I may have found the problem.

I need the native iOS Apple Pay button for my application. So I'm using the [pay_ios](https://pub.dev/packages/pay_ios) library and I realised that the accessibility focus is stuck on this button.

So I quickly recreated the plugin with a FlutterPlatformView and I have the same problem (so I don't think it's due to the package used).

Then I tried changing the semantics directly in the native part, but the problem is still there.

Finally, I put an ExludeSemantics around the UIKitView, which I then wrapped with a Semantics widget and that's when the problem was solved.

I really think there's a problem in the UiKitView widget about semantics.

In my logs, a debug message appears when the focus is on the UiKitView and it is blocked:
 
`[Accessibility] Went all the way up the container chain from @, but could not find any container that was one of the ordered children of @.  This may be acceptable if it happened right around a layout change, but it would be best to double check by swiping left/right to see if you can get to all elements.`

It's as if the UiKitView widget wasn't visible in the semantics tree and so VoiceOver was mixing up the widgets and crashing. 

I hope I've made myself clear, but please don't hesitate if you need any further explanation. 

### Expected results

When you slide the cursor to the left, your attention is drawn to the next widget.

### Actual results

When swiping left, when the focus goes on UiKitView, it remains blocked and the error appears in the log. 

### Code sample

```dart
return Scaffold(
      body: Center(
          child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          const Text("first example"),
          const Text("second example"),
          const Text("third example"),
          Container(
            color: Colors.red,
            child: RawApplePayButton(
              style: ApplePayButtonStyle.black,
              type: ApplePayButtonType.plain,
            ),
          ),
          const Text("last example"),
        ],
      )),
    );
```

### Screenshots or Video

https://github.com/flutter/flutter/assets/72022928/822aa667-79da-4c12-921e-900342c55298

### Logs

[Accessibility] Went all the way up the container chain from @, but could not find any container that was one of the ordered children of @.  This may be acceptable if it happened right around a layout change, but it would be best to double check by swiping left/right to see if you can get to all elements.

### Flutter Doctor output

• Flutter version 3.13.4 on channel stable at /Users/romain_greaume/fvm/versions/3.13.4
    ! Warning: `dart` on your path resolves to /opt/homebrew/Cellar/dart/3.0.7/libexec/bin/dart, which is not inside your current Flutter SDK checkout at /Users/romain_greaume/fvm/versions/3.13.4.
      Consider adding /Users/romain_greaume/fvm/versions/3.13.4/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 367f9ea16b (13 days ago), 2023-09-12 23:27:53 -0500
    • Engine revision 9064459a8b
    • Dart version 3.1.2
    • DevTools version 2.25.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at /Users/romain_greaume/Library/Android/sdk
    • Platform android-33, build-tools 33.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] IntelliJ IDEA Ultimate Edition (version 2023.1.4)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 75.1.3
    • Dart plugin version 231.9161.14

[✓] VS Code (version 1.82.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.72.0

[✓] Connected device (3 available)
    • iPhone 14 (mobile) • 968A4248-A23D-468B-9605-32E3C6ABD134 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
    • macOS (desktop)    • macos                                • darwin-arm64   • macOS 13.5 22G74 darwin-arm64
    • Chrome (web)       • chrome                               • web-javascript • Google Chrome 116.0.5845.187

[✓] Network resources
    • All expected network resources are available.

**Comment by darshankawar** [comment]
Thanks for the detailed report @romgrm 


> So I quickly recreated the plugin with a FlutterPlatformView and I have the same problem (so I don't think it's due to the package used).

Can you provide us the complete code for platformview as well so that we can run it thoroughly ?

The code sample you shared has `RawApplePayButton` which I think is coming from pay_ios library as you mentioned.

**Comment by romgrm** [comment]
Hello @darshankawar :) 

Thanks for your quick reply. 

I made a repo with the custom plugin: https://github.com/romgrm/UiKitView_A11Y_Issue

Very basic, but you can easily see the problem with UiKitView. 

Feel free to edit and don't hesitate if you have any questions 👍

**Comment by darshankawar** [comment]
Thanks for the update. Using the repro, I do see the same behavior as reported.

<details>
<summary> stable, master flutter doctor -v </summary>

```
[!] Flutter (Channel stable, 3.13.5, on macOS 12.2.1 21D62 darwin-x64, locale
    en-GB)
    • Flutter version 3.13.5 on channel stable at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 12fccda598 (32 hours ago), 2023-09-19 13:56:11 -0700
    • Engine revision bd986c5ed2
    • Dart version 3.1.2
    • DevTools version 2.25.0
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

[!] Flutter (Channel master, 3.15.0-10.0.pre.12, on macOS 12.2.1 21D62
    darwin-x64, locale en-GB)
    • Flutter version 3.15.0-10.0.pre.12 on channel master at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision bea63d7c81 (5 hours ago), 2023-09-26 20:46:25 -0400
    • Engine revision 0a8ad236e4
    • Dart version 3.2.0 (build 3.2.0-199.0.dev)
    • DevTools version 2.28.0-dev.12
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/dhs/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13C100
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] IntelliJ IDEA Ultimate Edition (version 2021.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 65.1.4
    • Dart plugin version 213.7228

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.29.0

[✓] Connected device (3 available)
    • Darshan's iphone (mobile) • 21150b119064aecc249dfcfe05e259197461ce23 • ios
      • iOS 15.3.1 19D52
    • macOS (desktop)           • macos                                    •
      darwin-x64     • macOS 12.2.1 21D62 darwin-x64
    • Chrome (web)              • chrome                                   •
      web-javascript • Google Chrome 109.0.5414.119

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
      
[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.



```

</details>

**Comment by cyanglaz** [comment]
Similar to https://github.com/flutter/flutter/issues/93024 which was supposed to be fixed by https://github.com/flutter/engine/pull/29531

It seems like a regression, I will bisect

**Comment by cyanglaz** [comment]
Bisected to https://github.com/flutter/flutter/pull/131958, the engine roll is from [75684126921c9b5040edc1ab7e6ba03f8af221ee](https://github.com/flutter/engine/commit/75684126921c9b5040edc1ab7e6ba03f8af221ee) to [c1f977f72b68da673ec493ab5df0beacacc7a88b](https://github.com/flutter/engine/commit/c1f977f72b68da673ec493ab5df0beacacc7a88b)

It is possibly https://github.com/flutter/engine/commit/b8332e3c4870512a941d685d7d375b1d2d62983c. I will bisect the engine

**Comment by cyanglaz** [comment]
https://github.com/flutter/engine/commit/b8332e3c4870512a941d685d7d375b1d2d62983c caused the regression. I will look into it.
